packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ArrayConstructor } from '@lwc/shared';
+/**
+ * Issue #1545 - Writing a custom util to convert NodeList and HTMLCollection instances to array.
+ * This is to aviod conflict with some legacy third party libraries like prototype.js
+ * that patch Array.prototype and are incapable of handling NodeList and HTMLCollection.
+ */
+export function arrayFromCollection<T extends Node, K extends Element>(
+    collection: NodeListOf<T> | HTMLCollectionOf<K>
+): Array<T | K> {
+    const size = collection.length;
+    const cloned: T[] | K[] = new ArrayConstructor(size);
```

Comment:
This method performs better than instantiating an empty array and calling ArrayPush.
https://jsperf.com/new-array-vs-push-lwc
<img width="974" alt="Screen Shot 2019-10-01 at 1 51 29 PM" src="https://user-images.githubusercontent.com/931465/65999737-dc8a1980-e452-11e9-8e5a-7dd1afb6497d.png">